// seatsio/dist/src/Events/CreateEventParams.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateEventParams = void 0;
var AbstractEventParams_1 = require("./AbstractEventParams");
var CreateEventParams = /** @class */ (function (_super) {
    __extends(CreateEventParams, _super);
    function CreateEventParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CreateEventParams.prototype.withChannels = function (channels) {
        this.channels = channels;
        return this;
    };
    CreateEventParams.prototype.withForSaleConfig = function (forSaleConfig) {
        this.forSaleConfig = forSaleConfig;
        return this;
    };
    CreateEventParams.prototype.withDate = function (date) {
        this.date = date;
        return this;
    };
    return CreateEventParams;
}(AbstractEventParams_1.AbstractEventParams));
exports.CreateEventParams = CreateEventParams;
