// seatsio/dist/src/Events/StatusChangeRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusChangeRequest = void 0;
var StatusChangeRequest = /** @class */ (function () {
    function StatusChangeRequest() {
        this.type = StatusChangeRequest.TYPE_CHANGE_STATUS_TO;
        this.eventKey = null;
        this.objectOrObjects = null;
        this.status = null;
        this.holdToken = null;
        this.orderId = null;
        this.keepExtraData = null;
        this.ignoreChannels = null;
        this.channelKeys = null;
        this.allowedPreviousStatuses = null;
        this.rejectedPreviousStatuses = null;
        this.resaleListingId = null;
        this.season = null;
    }
    StatusChangeRequest.prototype.withType = function (type) {
        this.type = type;
        return this;
    };
    StatusChangeRequest.prototype.withEventKey = function (eventKey) {
        this.eventKey = eventKey;
        return this;
    };
    StatusChangeRequest.prototype.withObjects = function (objectOrObjects) {
        this.objectOrObjects = objectOrObjects;
        return this;
    };
    StatusChangeRequest.prototype.withStatus = function (status) {
        this.status = status;
        return this;
    };
    StatusChangeRequest.prototype.withHoldToken = function (holdToken) {
        this.holdToken = holdToken;
        return this;
    };
    StatusChangeRequest.prototype.withOrderId = function (orderId) {
        this.orderId = orderId;
        return this;
    };
    StatusChangeRequest.prototype.withKeepExtraData = function (keepExtraData) {
        this.keepExtraData = keepExtraData;
        return this;
    };
    StatusChangeRequest.prototype.withIgnoreChannels = function (ignoreChannels) {
        this.ignoreChannels = ignoreChannels;
        return this;
    };
    StatusChangeRequest.prototype.withChannelKeys = function (channelKeys) {
        this.channelKeys = channelKeys;
        return this;
    };
    StatusChangeRequest.prototype.withAllowedPreviousStatuses = function (allowedPreviousStatuses) {
        this.allowedPreviousStatuses = allowedPreviousStatuses;
        return this;
    };
    StatusChangeRequest.prototype.withRejectedPreviousStatuses = function (rejectedPreviousStatuses) {
        this.rejectedPreviousStatuses = rejectedPreviousStatuses;
        return this;
    };
    StatusChangeRequest.prototype.withResaleListingId = function (resaleListingId) {
        this.resaleListingId = resaleListingId;
        return this;
    };
    StatusChangeRequest.prototype.withSeasonKey = function (seasonKey) {
        this.season = seasonKey;
        return this;
    };
    StatusChangeRequest.TYPE_RELEASE = 'RELEASE';
    StatusChangeRequest.TYPE_CHANGE_STATUS_TO = 'CHANGE_STATUS_TO';
    StatusChangeRequest.TYPE_OVERRIDE_SEASON_STATUS = 'OVERRIDE_SEASON_STATUS';
    StatusChangeRequest.TYPE_USE_SEASON_STATUS = 'USE_SEASON_STATUS';
    return StatusChangeRequest;
}());
exports.StatusChangeRequest = StatusChangeRequest;
