// seatsio/dist/src/SeatsioClient.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeatsioClient = void 0;
var axios_1 = __importDefault(require("axios"));
var Accounts_1 = require("./Accounts/Accounts");
var Charts_1 = require("./Charts/Charts");
var errorInterceptor_1 = require("./errorInterceptor");
var EventLog_1 = require("./EventLog/EventLog");
var Events_1 = require("./Events/Events");
var HoldTokens_1 = require("./HoldTokens/HoldTokens");
var ChartReports_1 = require("./Reports/ChartReports");
var EventReports_1 = require("./Reports/EventReports");
var UsageReports_1 = require("./Reports/UsageReports");
var Seasons_1 = require("./Seasons/Seasons");
var TicketBuyers_1 = require("./TicketBuyers/TicketBuyers");
var Workspaces_1 = require("./Workspaces/Workspaces");
var TEN_SECONDS = 10000;
var SeatsioClient = /** @class */ (function () {
    function SeatsioClient(region, secretKey, workspaceKey, extraHeaders) {
        if (workspaceKey === void 0) { workspaceKey = undefined; }
        if (extraHeaders === void 0) { extraHeaders = {}; }
        this.client = axios_1.default.create(this._axiosConfig(region.url, secretKey, workspaceKey, extraHeaders));
        this._setupRequestListenerInterceptors();
        this.setMaxRetries(5);
        this.client.interceptors.response.use(function (response) { return response; }, exponentialBackoffInterceptor(this.client));
        this.errInterceptor = this.client.interceptors.response.use(function (response) { return response; }, errorInterceptor_1.errorResponseHandler);
        this.charts = new Charts_1.Charts(this.client);
        this.events = new Events_1.Events(this.client);
        this.workspaces = new Workspaces_1.Workspaces(this.client);
        this.holdTokens = new HoldTokens_1.HoldTokens(this.client);
        this.accounts = new Accounts_1.Accounts(this.client);
        this.chartReports = new ChartReports_1.ChartReports(this.client);
        this.eventReports = new EventReports_1.EventReports(this.client);
        this.usageReports = new UsageReports_1.UsageReports(this.client);
        this.seasons = new Seasons_1.Seasons(this.client, this);
        this.eventLog = new EventLog_1.EventLog(this.client, this);
        this.ticketBuyers = new TicketBuyers_1.TicketBuyers(this.client, this);
    }
    SeatsioClient.prototype._axiosConfig = function (baseUrl, secretKey, workspaceKey, extraHeaders) {
        var headers = __assign(__assign(__assign({}, extraHeaders), { 'X-Client-Lib': 'js' }), (workspaceKey ? { 'X-Workspace-Key': workspaceKey } : {}));
        var config = {
            baseURL: baseUrl,
            auth: {
                username: secretKey,
                password: ''
            },
            headers: headers,
            timeout: TEN_SECONDS,
            paramsSerializer: {
                indexes: null
            }
        };
        return config;
    };
    SeatsioClient.prototype._setupRequestListenerInterceptors = function () {
        var _this = this;
        this.client.interceptors.request.use(function (config) {
            if (_this.requestListener) {
                config.listener = _this.requestListener();
                config.listener.onRequestStarted();
            }
            return config;
        });
        this.client.interceptors.response.use(function (response) {
            if (response.config.listener) {
                response.config.listener.onRequestEnded();
            }
            return response;
        }, function (response) {
            if (response.config.listener) {
                response.config.listener.onRequestEnded();
            }
            return Promise.reject(response);
        });
    };
    SeatsioClient.prototype.setRequestListener = function (requestListener) {
        this.requestListener = requestListener;
        return this;
    };
    SeatsioClient.prototype.setMaxRetries = function (maxRetries) {
        // @ts-ignore
        this.client.maxRetries = maxRetries;
        return this;
    };
    return SeatsioClient;
}());
exports.SeatsioClient = SeatsioClient;
function exponentialBackoffInterceptor(axios) {
    return function (error) {
        var _a;
        if (((_a = error.response) === null || _a === void 0 ? void 0 : _a.status) !== 429) {
            return Promise.reject(error);
        }
        var config = error.config;
        if (!config) {
            return Promise.reject(error);
        }
        config.__retryCount = config.__retryCount || 0;
        if (config.__retryCount >= axios.maxRetries) {
            return Promise.reject(error);
        }
        var backoff = new Promise(function (resolve) {
            var waitTime = Math.pow(2, config.__retryCount + 2) * 100;
            config.__retryCount++;
            setTimeout(function () { return resolve(); }, waitTime);
        });
        return backoff.then(function () { return axios.request(config); });
    };
}
