// seatsio/dist/src/Seasons/CreateSeasonParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSeasonParams = void 0;
var CreateSeasonParams = /** @class */ (function () {
    function CreateSeasonParams() {
    }
    CreateSeasonParams.prototype.key = function (key) {
        this._key = key;
        return this;
    };
    CreateSeasonParams.prototype.name = function (name) {
        this._name = name;
        return this;
    };
    CreateSeasonParams.prototype.numberOfEvents = function (numberOfEvents) {
        this._numberOfEvents = numberOfEvents;
        return this;
    };
    CreateSeasonParams.prototype.eventKeys = function (eventKeys) {
        this._eventKeys = eventKeys;
        return this;
    };
    CreateSeasonParams.prototype.tableBookingConfig = function (tableBookingConfig) {
        this._tableBookingConfig = tableBookingConfig;
        return this;
    };
    CreateSeasonParams.prototype.channels = function (channels) {
        this._channels = channels;
        return this;
    };
    CreateSeasonParams.prototype.forSaleConfig = function (forSaleConfig) {
        this._forSaleConfig = forSaleConfig;
        return this;
    };
    CreateSeasonParams.prototype.forSalePropagated = function (forSalePropagated) {
        this._forSalePropagated = forSalePropagated;
        return this;
    };
    CreateSeasonParams.prototype.objectCategories = function (objectCategories) {
        this._objectCategories = objectCategories;
        return this;
    };
    CreateSeasonParams.prototype.categories = function (categories) {
        this._categories = categories;
        return this;
    };
    return CreateSeasonParams;
}());
exports.CreateSeasonParams = CreateSeasonParams;
